sdk/include/platform/sunburst/platform-pwm.hh

```diff
@@ -0,0 +1,66 @@
+#pragma once
+#include <debug.hh>
+#include <stdint.h>
+
+/**
+ * A driver for Sonata's PWM.
+ *
+ * Documentation source can be found at:
+ * https://github.com/lowRISC/sonata-system/blob/97a525c48f7bf051b999d0178dba04859819bc5e/doc/ip/pwm.md
+ *
+ * Rendered documentation is served from:
+ * https://lowrisc.github.io/sonata-system/doc/ip/pwm.html
+ */
+struct SonataPwm
+{
+	/**
+	 * Flag to set when debugging the driver for UART log messages.
+	 */
+	static constexpr bool DebugPwm = false;
+
+	/**
+	 * Helper for conditional debug logs and assertions.
+	 */
+	using Debug = ConditionalDebug<DebugPwm, "PWM">;
+
+	/**
+	 * The number of PWMs that are initialised.
+	 */
+	static constexpr size_t PwmCount = 1;
+
+	/**
+	 * The maximum supported period for a PWM block wave.
+	 */
+	static constexpr uint32_t MaxPwmPeriod = 255;
+
+	/**
+	 * The PWM devices available on Sonata.
+	 */
+	struct PwmRegisters
+	{
+		/**
+		 * The duty cycle of the PWM wave, represented as a width counter. That
+		 * is, the number of clock cycles for which the signal will be on. The
+		 * duty cycle as a percentage is (duty cycle / period) * 100.
+		 */
+		uint32_t dutyCycle;
+
+		/**
+		 * The period (width) of the block wave.
```

Comment:
A little more here would be nice. Something like: "The period of the output, set with the number of clock cycles a period should last."